pilot/pkg/model/jwks_resolver.go

```diff
@@ -138,28 +138,40 @@ func NewJwksResolver(evictionDuration, refreshInterval time.Duration) *JwksResol
 		},
 	}
 
-	caCert, err := ioutil.ReadFile(publicRootCABundlePath)
-	if err == nil {
-		caCertPool := x509.NewCertPool()
-		caCertPool.AppendCertsFromPEM(caCert)
-		ret.secureHTTPClient = &http.Client{
-			Timeout: jwksHTTPTimeOutInSec * time.Second,
-			Transport: &http.Transport{
-				DisableKeepAlives: true,
-				TLSClientConfig: &tls.Config{
-					RootCAs: caCertPool,
-				},
-			},
-		}
-	}
-
 	atomic.StoreUint64(&ret.refreshJobKeyChangedCount, 0)
 	atomic.StoreUint64(&ret.refreshJobFetchFailedCount, 0)
 	go ret.refresher()
 
 	return ret
 }
 
+// InitializeSecureClient sets up an HTTPS client to be used for resolving jwks_uri
+func (r *JwksResolver) InitializeSecureClient(usePublicRootCABundle bool, customRootCA string) {
+	caCertPool := x509.NewCertPool()
+	if usePublicRootCABundle {
```

Comment:
could we just put public ca bundle by default ?

Reply:
Done.